src/System.Threading.Tasks/tests/TaskFactory/TaskFactory_FromAsyncTests.cs

```diff
@@ -23,36 +23,23 @@ public static void RunAPMFactoryTests()
             string check;
             object stateObject = new object();
 
-
             // Exercise void overload that takes IAsyncResult instead of StartMethod
             t = Task.Factory.FromAsync(fac.StartWrite("", 0, 0, null, null), delegate (IAsyncResult iar) { });
             t.Wait();
             check = fac.ToString();
-            if (check.Length != 0)
-            {
-                Assert.True(false, string.Format("RunAPMFactoryTests:    > FAILED on Write1 -- expected empty fac."));
-            }
+            Assert.Equal(0, check.Length);
 
             //CreationOption overload
             t = Task.Factory.FromAsync(fac.StartWrite("", 0, 0, null, null), delegate (IAsyncResult iar) { }, TaskCreationOptions.None);
             t.Wait();
             check = fac.ToString();
-            if (check.Length != 0)
-            {
-                Assert.True(false, string.Format("RunAPMFactoryTests:    > FAILED on Write2 -- expected empty fac."));
-            }
+            Assert.Equal(0, check.Length);
 
             // Exercise 0-arg void option
             t = Task.Factory.FromAsync(fac.StartWrite, fac.EndWrite, stateObject);
             t.Wait();
-            if (check.Length != 0)
-            {
-                Assert.True(false, string.Format("RunAPMFactoryTests:    > FAILED on Write3 -- expected empty fac."));
-            }
-            else if (((IAsyncResult)t).AsyncState != stateObject)
-            {
-                Assert.True(false, string.Format("RunAPMFactoryTests:    > FAILED on Write3 -- state object not stored correctly."));
-            }
+            Assert.Equal(0, check.Length);
+            Assert.Equal(stateObject, ((IAsyncResult)t).AsyncState);
```

Comment:
These checks for the state object should probably be Assert.Same rather than Assert.Equal, though in practice it doesn't matter given that it's an Object.